c10/util/MaybeOwned.h

```diff
@@ -103,6 +103,19 @@ class MaybeOwned final {
     }
     return isBorrowed_ ? borrow_ : &own_;
   }
+
+  // If borrowed, copy the underlying T. If owned, move from
+  // it. borrowed/owned state remains the same, and either we
+  // reference the same borrow as before or we are an owned moved-from
+  // T.
+  T operator*() && {
+    if (isBorrowed_) {
+      TORCH_INTERNAL_ASSERT_DEBUG_ONLY(borrow_ != nullptr);
```

Comment:
No action needed, but curious the motivation behind checking this on derefs rather than on construction/assignment

Reply:
Default construction creates a null borrowed MaybeOwned to simplify initialization inside an if statement.